packages/cloudvision-grpc-web/src/grpc/index.ts

```diff
@@ -57,10 +59,12 @@ export function fromGrpcInvoke<Req extends grpc.ProtobufMessage, Res extends grp
     },
   };
 
-  grpc.invoke(methodDescriptor, rpcOptions);
+  const subscribe = grpc.invoke(methodDescriptor, rpcOptions);
+
+  const dataObs$ = new GrpcSubjectWrapper(dataSubject, subscribe.close);
 
   return {
-    data: dataSubject.asObservable(),
+    data: dataObs$,
```

Comment:
This is the entire solution
```
data: dataSubject.pipe(finalize(subscribe.close));
```

Reply:
The issue here is that unsubscribe is called atleast twice, one when the `takeUntil` in bufferOnce unsubscribes, and the final one being the last listener unsubscribing. This finalizer is called every time there’s an unsubscribe which prematurely closes the grpc request. 

I need to keep track of listeners and only call unsubscribe when the last listener is gone. How would I go about it here?